diff --git a/target/linux/ipq60xx/files/arch/arm64/boot/dts/qcom/ipq6000-cmiot.dtsi b/target/linux/ipq60xx/files/arch/arm64/boot/dts/qcom/ipq6000-cmiot.dtsi
--- a/target/linux/ipq60xx/files/arch/arm64/boot/dts/qcom/ipq6000-cmiot.dtsi
+++ b/target/linux/ipq60xx/files/arch/arm64/boot/dts/qcom/ipq6000-cmiot.dtsi
@@ -60,10 +60,74 @@
  &qpic_nand {
    status = "okay";

    nand@0 {
        reg = <0>;
@@ -71,7 +135,71 @@
        nand-ecc-step-size = <512>;
        nand-bus-width = <8>;

-       partitions {
-           compatible = "qcom,smem-part";
-       };
+       partitions {
+           compatible = "fixed-partitions";
+           #address-cells = <1>;
+           #size-cells = <1>;
+
+           partition@0 {
+               label = "0:sbl1";
+               reg = <0x00000000 0x00180000>;
+               read-only;
+           };
+
+           partition@180000 {
+               label = "0:mibib";
+               reg = <0x00180000 0x00100000>;
+               read-only;
+           };
+
+           partition@280000 {
+               label = "0:bootconfig";
+               reg = <0x00280000 0x00080000>;
+               read-only;
+           };
+
+           partition@300000 {
+               label = "0:bootconfig1";
+               reg = <0x00300000 0x00080000>;
+               read-only;
+           };
+
+           partition@380000 {
+               label = "0:qsee";
+               reg = <0x00380000 0x00380000>;
+               read-only;
+           };
+
+           partition@700000 {
+               label = "0:qsee_1";
+               reg = <0x00700000 0x00380000>;
+               read-only;
+           };
+
+           partition@a80000 {
+               label = "0:devcfg";
+               reg = <0x00A80000 0x00080000>;
+               read-only;
+           };
+
+           partition@b00000 {
+               label = "0:devcfg_1";
+               reg = <0x00B00000 0x00080000>;
+               read-only;
+           };
+
+           partition@b80000 {
+               label = "0:rpm";
+               reg = <0x00B80000 0x00080000>;
+               read-only;
+           };
+
+           partition@c00000 {
+               label = "0:rpm_1";
+               reg = <0x00C00000 0x00080000>;
+               read-only;
+           };
+
+           partition@c80000 {
+               label = "0:cdt";
+               reg = <0x00C80000 0x00080000>;
+               read-only;
+           };
+
+           partition@d00000 {
+               label = "0:cdt_1";
+               reg = <0x00D00000 0x00080000>;
+               read-only;
+           };
+
+           partition@d80000 {
+               label = "0:appsblenv";
+               reg = <0x00D80000 0x00080000>;
+               read-only;
+           };
+
+           partition@e00000 {
+               label = "0:appsbl";
+               reg = <0x00E00000 0x00180000>;
+               read-only;
+           };
+
+           partition@f80000 {
+               label = "0:appsbl_1";
+               reg = <0x00F80000 0x00180000>;
+               read-only;
+           };
+
+           partition@1100000 {
+               label = "0:art";
+               reg = <0x01100000 0x00080000>;
+               read-only;
+           };
+
+           partition@1180000 {
+               label = "ubi";
+               reg = <0x01180000 0x06E80000>;
+           };
+       };
    };
  };
